contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,335 @@
+// Package mgo provides functions and types which allow tracing of the MGO MongoDB client (https://github.com/globalsign/mgo)
+package mgo // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/globalsign/mgo"
+
+import (
+	"github.com/globalsign/mgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Dial opens a connection to a MongoDB server and ties it to the
+// given context for tracing MongoDB calls.
+func Dial(url string, opts ...MongoOption) (*Session, error) {
+	session, err := mgo.Dial(url)
+	s := &Session{
+		Session: session,
+	}
+
+	defaults(&s.cfg)
+	for _, fn := range opts {
+		fn(&s.cfg)
+	}
+
+	return s, err
+}
+
+// Session wraps a Session with the Context and Config information
+// needed to support tracing.
+type Session struct {
+	*mgo.Session
+
+	cfg mongoConfig
+}
+
+func newChildSpanFromContext(config mongoConfig) ddtrace.Span {
+	name := "mongodb.query"
+	span, _ := tracer.StartSpanFromContext(
+		config.ctx,
+		name,
+		tracer.SpanType(ext.SpanTypeMongoDB),
+		tracer.ServiceName(config.serviceName),
+		tracer.ResourceName("mongodb.query"))
+
+	return span
+}
+
+// Run invokes and traces Session.Run
+func (s *Session) Run(cmd interface{}, result interface{}) (err error) {
+	span := newChildSpanFromContext(s.cfg)
+	err = s.Session.Run(cmd, result)
+	span.Finish(tracer.WithError(err))
+	return
+}
+
+// Database wraps a Database with the Context and Config information
+// needed to support tracing.
+type Database struct {
+	*mgo.Database
+	cfg mongoConfig
+}
+
+// DB invokes Session.DB and wraps it with the context and config
+// data from `WrapSession` that is needed to support tracing
```

Comment:
Let's not use this phrasing "context and config" anywhere. We shouldn't refer to any technical terms not visible in the function signature.
```go
// DB returns a new Database for this Session.
```
Then people can see what Database it is in its own godoc section.